multicast/hear.py

```diff
@@ -327,35 +297,33 @@ class HearUDPHandler(socketserver.BaseRequestHandler):
 	"""
 
 	def handle(self):
-		data = self.request[0].strip()
-		socket = self.request[1]
+		(data, sock) = self.request
 		print(str("{} SAYS: {} to {}").format(
-			self.client_address[0], str(data), "ALL"
+			self.client_address[0], data.strip(), "ALL"
 		))
 		if data is not None:
-			myID = str(socket.getsockname()[0])
-			print(
-				str("{me} HEAR: [{you} SAID {what}]").format(
-					me=myID, you=self.client_address, what=str(data)
-				)
-			)
-			print(
-				str("{me} SAYS [ HEAR [ {what} SAID {you} ] from {me} ]").format(
-					me=myID, you=self.client_address, what=str(data)
-				)
-			)
+			myID = str(sock.getsockname()[0])
+			if (_sys.stdout.isatty()):  # pragma: no cover
+				_sim_data_str = data.strip().replace('\r', '').replace('%', '%%')
+				print(str("{me} HEAR: [{you} SAID {what}]").format(
+					me=myID, you=self.client_address, what=str(_sim_data_str)
+				))
+				print(str("{me} SAYS [ HEAR [ {what} SAID {you} ] from {me} ]").format(
+					me=myID, you=self.client_address, what=str(_sim_data_str)
+				))
```

Comment:
LGTM, make a note about this on #134
I'll add these changes to the batch of changes.

Reply:
@coderabbitai, yes add the note to issue #134 regarding this change!

we'll be going with:

```python3
_sim_data_str = data.strip().replace('\r', '').replace('%', '%%')
print(f"{myID} HEAR: [{self.client_address} SAID {_sim_data_str}]")
print(f"{myID} SAYS [ HEAR [ {_sim_data_str} SAID {self.client_address} ] from {myID} ]")
```
to avoid formatting and use the newer python f-strings